cowprotocol/cowswap
# Update logic to properly handle buy order changes

# Summary

This PR updates the behavior of the widget a bit

If its a **BUY** order (user modified the output field), then any change to a price will update the input field
Also if he changes the currency, the input field value will be re-calculated

If its a **SELL** order (user modified the input field), then any change to price will update the output field ...
Changes to currency will modify the output field value

# To test
1. Go to limit orders
2. Modify the output field so it becomes BUY order
3. Modify the price
4. The input amount should be re-calculated
5. Change the tokens, the rate should be updated and the new input amount calculated
